cas-server-support-spnego/src/main/java/org/jasig/cas/support/spnego/web/flow/client/BaseSpnegoKnownClientSystemsFilterAction.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.support.spnego.web.flow.client;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.jasig.cas.web.support.WebUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Abstract class for defining a simple binary filter to determine whether a
+ * given client system should be prompted for SPNEGO / KRB / NTLM credentials.
+ *
+ * Envisioned implementations would include LDAP and DNS based determinations,
+ * but of course others may have value as well for local architectures.
+ *
+ * @author Sean Baker sean.baker@usuhs.edu
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public class BaseSpnegoKnownClientSystemsFilterAction extends AbstractAction {
+    private static final int DEFAULT_TIMEOUT = 5000;
+
+    /** Logger instance. **/
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /** Pattern of ip addresses to check. **/
+    private Pattern ipsToCheckPattern;
```

Comment:
Understood, but my understanding is, that can also be specified as a regex pattern, something like \w+\.\w+\.\w+\.\w+\./\d\d

An exact string would require exact matches and that will not work so well against a range of IP addresses. (The pattern would then need to into a set)